manager/orchestrator/task.go

```diff
@@ -93,6 +93,32 @@ func IsTaskDirty(s *api.Service, t *api.Task) bool {
 		(t.Endpoint != nil && !reflect.DeepEqual(s.Spec.Endpoint, t.Endpoint.Spec))
 }
 
+// IsTaskPlacementConstraintsOnlyDirty checks if the Placement field alone
+// in the spec has changed.
+func IsTaskPlacementConstraintsOnlyDirty(s *api.Service, t *api.Task) bool {
```

Comment:
Needs a unit test IMHOIRL.

Reply:
Also IMO `IsTaskDirtyPlacementConstraintsOnly` would be a better name.

Reply:
Agreed on both.

Reply:
`areOnlyTaskPlacementConstraintsDirty`?